pkg/skaffold/gcp/auth.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2018 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gcp
+
+import (
+	"strings"
+
+	"github.com/docker/cli/cli/config/configfile"
+)
+
+// AutoConfigureGCRCredentialHelper automatically adds the `gcloud` credential helper
+// to docker's configuration.
+// This doesn't modify the ~/.docker/config.json. It's only in-memory
+func AutoConfigureGCRCredentialHelper(cf *configfile.ConfigFile, registry string) {
+	if registry != "gcr.io" && !strings.HasSuffix(registry, ".gcr.io") {
+		return
+	}
+
+	if cf.CredentialHelpers == nil {
+		cf.CredentialHelpers = make(map[string]string)
+	}
+
+	_, present := cf.CredentialHelpers[registry]
+	if !present {
+		cf.CredentialHelpers[registry] = "gcloud"
```

Comment:
Do we make sure gcloud is installed anywhere?

It's also possible to use the "gcr" credential helper which handles it's own authentication without gcloud. I'm not sure how common that is, though.

Reply:
I'm going to add the check indeed.

I'd rather ignore that "gcr" credential helper can be used since It's not the solution documented [here](
https://cloud.google.com/container-registry/docs/advanced-authentication)
Users can still use it if they want though.